AGENTS.md

```diff
@@ -0,0 +1,103 @@
+# Metal3 IP Address Manager - AI Agent Instructions
+
+Instructions for AI coding agents. For project overview, see [README.md](README.md).
+For contribution guidelines, see [CONTRIBUTING.md](CONTRIBUTING.md).
+
+## Repository Structure
+
+| Directory | Purpose |
+|-----------|---------|
+| `api/v1alpha1/` | CRD types (IPPool, IPClaim, IPAddress) - separate Go module |
+| `controllers/` | Reconciliation logic |
+| `ipam/` | Core IP allocation algorithms |
+| `internal/webhooks/` | Validation/defaulting webhooks |
+| `config/` | Kustomize manifests (CRDs, RBAC, webhooks) - auto-generated by `make generate`, rarely edit directly |
+| `hack/` | CI scripts (prefer Make targets locally) |
+
+## Testing Standards
+
+CI is managed by [Prow](https://github.com/metal3-io/project-infra/blob/main/prow/config/jobs/metal3-io/ip-address-manager.yaml).
+Run these locally before submitting PRs:
+
+**Make targets:**
+
+| Command | Purpose |
+|---------|---------|
+| `make test` | Full verification (generate + lint + unit) |
+| `make generate` | Regenerate code (DeepCopy, CRDs, RBAC, mocks) |
+| `make unit` | Unit tests (requires envtest - do not use `go test` directly) |
+| `make lint` | Go linting via golangci-lint (all modules) |
+| `make verify-boilerplate` | Check license headers |
+| `make modules` | Verify go.mod is tidy |
+
```

Comment:
question: why are we adding selective make targets here? why not all of them?
Also I don't see any section related to releasing should we add something related to that too?